pilot/pkg/networking/core/v1alpha3/route/route_cache.go

```diff
@@ -78,7 +78,7 @@ func (r *Cache) DependentConfigs() []model.ConfigKey {
 	configs := make([]model.ConfigKey, 0, len(r.Services)+len(r.VirtualServices)+
 		len(r.DelegateVirtualServices)+len(r.DestinationRules)+len(r.EnvoyFilterKeys))
 	for _, svc := range r.Services {
-		configs = append(configs, model.ConfigKey{Kind: gvk.ServiceEntry, Name: string(svc.Hostname), Namespace: svc.Attributes.Namespace})
+		configs = append(configs, model.ConfigKey{Kind: gvk.ServiceEntry, Name: svc.Attributes.Name, Namespace: svc.Attributes.Namespace})
```

Comment:
I donot think i understand this change, 

as from below, we use hostname to clear

```
	s.Cache.Clear(map[model.ConfigKey]struct{}{{
		Kind:      gvk.ServiceEntry,
		Name:      hostname,
		Namespace: namespace,
	}: {}})
```

Reply:
Yeah sorry for the confusion. I think the root cause is that namespace at key is not populated when inserting new entry.